rpcs3/Emu/Cell/SPUThread.h

```diff
@@ -212,22 +213,16 @@ struct spu_channel_t
 	// returns true on success
 	bool try_pop(u32& out)
 	{
-		const auto old = data.fetch_op([&](sync_var_t& data)
+		const auto old = data.fetch_op([](sync_var_t& data)
 		{
-			if (data.count)
-			{
-				data.wait = false;
-				out = data.value;
-			}
-			else
-			{
-				data.wait = true;
-			}
-
-			data.count = false;
-			data.value = 0; // ???
+			sync_var_t t;
+			*reinterpret_cast<u64*>(&t) = 0;
```

Comment:
Why this atrocity?

Reply:
Way to cause the compiler to generate the same code with fewer instructions.
Micro-optimization - but as it accounted for a large percentage of run time, I think its needed.

Reply:
@Farseer2 do you actually have any evidence that this is an optimization?  ALL major compilers optimize it to simple `xor`: https://godbolt.org/g/PqQEuY
Looks like introducing smelly unmaintainable code just for the sake of it (what if t changes size in the future? Or aligning? Or padding? This code just looks incorrect).

Reply:
Oh, thought you meant the entire change, not this single line - it is a left over from when I tried it with the reference, and not a new var.
True, it is a mistake, but not one done in malice - this is what the review is for, no need to be aggressive.

Reply:
@Farseer2 ah sorry if I somehow offended you. Wasn't my intention.